core/templates/services/prevent-page-unload-event.service.ts

```diff
@@ -39,6 +39,9 @@ export class PreventPageUnloadEventService {
   }
 
   removeListener(): void {
+    if (!this.listenerActive) {
```

Comment:
Why this check is needed? [WHy are we expecting some directives to remove listeners without adding one? or Am I missing something?]

Reply:
Yes, This is just in case some listeners keep calling the removelistener when there is actually no listener present.